File: springboot/src/main/java/com/edufelip/meer/dto/ThriftStoreDto.java
package com.edufelip.meer.dto;

import java.time.Instant;
import java.util.List;

public record ThriftStoreDto(
    java.util.UUID id,
    String name,
    String tagline,
    String coverImageUrl,
    String addressLine,
    Double latitude,
    Double longitude,
    String openingHours,
    String facebook,
    String instagram,
    String website,
    String phone,
    String whatsapp,
    List<String> categories,
    Double rating,
    Integer reviewCount,
    Integer myRating,
    Double distanceMeters,
    Integer walkTimeMinutes,
    String neighborhood,
    String badgeLabel,
    Boolean isFavorite,
    String description,
    List<GuideContentDto> contents,
    List<StoreImageDto> images,
    Instant createdAt) {}

File: springboot/src/main/kotlin/com/edufelip/meer/dto/ThriftStoreDto.kt
package com.edufelip.meer.dto

data class ThriftStoreDto(
    val id: Int,
    val name: String,
    val tagline: String,
    val coverImageUrl: String,
    val galleryUrls: List<String>?,
    val addressLine: String,
    val latitude: Double?,
    val longitude: Double?,
    val mapImageUrl: String?,
    val openingHours: String,
    val openingHoursNotes: String?,
    val facebook: String?,
    val instagram: String?,
    val website: String?,
    val whatsapp: String?,
    val categories: List<String>,
    val distanceKm: Double?,
    val walkTimeMinutes: Int?,
    val neighborhood: String?,
    val badgeLabel: String?,
    val isFavorite: Boolean?,
    val description: String?,
    val contents: List<GuideContentDto>? = null
)
